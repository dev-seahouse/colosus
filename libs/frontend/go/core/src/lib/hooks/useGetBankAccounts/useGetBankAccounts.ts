import type {
  InvestorBrokerageGetBankAccountsForPartyResponseDto,
  StandardError,
} from '@bambu/api-client';
import {
  hasNoAccountFoundError,
  TransactInvestorAuthenticatedBrokerageApi,
} from '@bambu/api-client';
import type { QueryClient } from '@tanstack/react-query';
import { useQuery } from '@tanstack/react-query';
import type { QueryArgs } from '../../types/utils';

export const getBankAccountsQuery = {
  queryKey: ['getBankAccounts'],
  queryFn: fetchBankAccounts,
  retry: (count: number, error: StandardError) => {
    if (hasNoAccountFoundError(error)) return false;
    return count <= 5;
  },
  retryDelay: 2000,
  refecthOnWindowFocus: false,
};
export function useGetBankAccounts<
  T = InvestorBrokerageGetBankAccountsForPartyResponseDto
>(
  queryOptions?: QueryArgs<
    InvestorBrokerageGetBankAccountsForPartyResponseDto,
    T
  >
) {
  return useQuery({
    ...getBankAccountsQuery,
    ...queryOptions,
  });
}
export const getBankAccountsLoader = (queryClient: QueryClient) => {
  return async () => {
    const query = getBankAccountsQuery;
    return (
      queryClient.getQueryData(query.queryKey) ??
      (await queryClient.fetchQuery(query))
    );
  };
};

async function fetchBankAccounts() {
  const bankAccountsApi = new TransactInvestorAuthenticatedBrokerageApi();
  const res = await bankAccountsApi.getBankAccounts();
  return res.data;
}
export default useGetBankAccounts;
