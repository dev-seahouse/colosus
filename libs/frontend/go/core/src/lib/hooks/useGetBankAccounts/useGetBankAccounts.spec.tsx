import useGetBankAccounts from './useGetBankAccounts';
import {
  queryClientWrapper,
  renderHook,
  waitFor,
} from '@bambu/react-test-utils';

describe('useGetBankAccounts', () => {
  it('should render successfully', async () => {
    const { result } = renderHook(() => useGetBankAccounts(), {
      wrapper: queryClientWrapper,
    });
    await waitFor(() => expect(result.current.isSuccess).toBe(true));
    expect(result.current.data).toMatchSnapshot();
  });
});
