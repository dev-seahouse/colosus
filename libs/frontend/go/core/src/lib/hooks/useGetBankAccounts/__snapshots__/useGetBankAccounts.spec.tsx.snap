// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`useGetBankAccounts > should render successfully 1`] = `
[
  {
    "accountNumber": "55779911",
    "addedAt": "2021-01-01T00:00:00.000Z",
    "clientReference": "1944a59713",
    "countryCode": "GB",
    "currency": "GBP",
    "deactivatedAt": "2021-01-01T00:00:00.000Z",
    "id": "bnk-33wdqmffe22aq6",
    "name": "C Kent & L Lane",
    "partyId": "pty-33wds6i4i242wc",
    "sortCode": "20-00-00",
  },
]
`;
